### PR TITLE
[2.x] Merge librarymanagement and sbt-coursier

### PR DESCRIPTION
**Pull request by adpi2**
For sbt 2.x only, we merge [sbt/librarymanagement](https://github.com/sbt/librarymanagement) and [coursier/sbt-coursier](https://github.com/coursier/sbt-coursier) into sbt main repo.

The repo contains 3 new folders: `lm-core`, `lm-ivy` and `lm-coursier`.

From [coursier/sbt-coursier](https://github.com/coursier/sbt-coursier), I removed `sbt-coursier` and `sbt-lm-coursier` but I kept some of the scripted tests that were not copies of `dependency-management/*`. I moved them to [sbt-app/src/sbt-test/lm-coursier/*](https://github.com/sbt/sbt/tree/e47a4e196d9750f501f2b94d5f9510daf1328de3/sbt-app/src/sbt-test/lm-coursier) along with other scripted tests. The `lm-coursier/metadata` submodule is required to run those tests, see [scripts/lm-coursier.sh](https://github.com/sbt/sbt/blob/e47a4e196d9750f501f2b94d5f9510daf1328de3/scripts/lm-coursier-ci.sh).

In the CI workflow, I added job-type 10, which runs the lm-coursier tests and scripted. It runs on ubuntu-latest and windows-latest (about 10 min and 14 min).

Also added a `NOTICE` file for license and copyright.

Ping @alexarchambault

### CODE REVIEW

File: bin/fixpreloaded.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+rm -rf $HOME/.sdkman/candidates/sbt/current/lib/local-preloaded/com.google.protobuf/protobuf-java/3.3.1
+rm -rf $HOME/.sdkman/candidates/sbt/current/lib/local-preloaded/org.scala-lang/scala-compiler/2.12.6
+
+rm -rf $HOME/.sbt/preloaded/org.scala-lang.modules/scala-parser-combinators_2.12/1.0.4
+rm -rf $HOME/.sbt/preloaded/org.scala-lang.modules/scala-parser-combinators_2.12/1.0.5
+rm -rf $HOME/.sbt/preloaded/org.scala-lang.modules/scala-xml_2.12/1.0.6
+rm -rf $HOME/.sbt/preloaded/com.typesafe/config/1.2.0
+rm -rf $HOME/.sbt/preloaded/com.typesafe/ssl-config-core_2.12/0.2.2
+rm -rf $HOME/.sbt/preloaded/com.google.protobuf/protobuf-java/3.3.1
+rm -rf $HOME/.sbt/preloaded/org.scala-lang/scala-compiler/2.12.6
```
Comment (adpi2) [inline bin/fixpreloaded.sh]: Do we need to keep this file?

Comment (eed3si9n) [reply bin/fixpreloaded.sh]: Probably not.

**Comment by eed3si9n** [comment]
We probably want to keep the git history though.

**Review by eed3si9n — APPROVED** [review]
Thanks!!

**Pull request merged**